### ISSUE TITLE
Bump Wazuh dashboard to OpenSearch Dashboards 2.8.0

### ISSUE THREAD

**Issue by rauldpm** [wazuh/wazuh-packages#2392]
## Description

It is necessary to adapt the Wazuh dashboard to version 2.8.0 of OpenSearch Dashboards
Request: https://github.com/wazuh/internal-devel-requests/issues/194

## Tasks

- [x] Bump OpenSearch Dashboards version https://github.com/wazuh/wazuh-packages/blob/55cfe35dfa163ac951884c22a390fe1838651766/stack/dashboard/base/builder.sh#L19
- [x] Build the Wazuh dashboard app for 4.6.0-2.8.0 locally -> tag [v4.6.0-2.8.0-alpha1](https://github.com/wazuh/wazuh-kibana-app/tree/v4.6.0-2.8.0-alpha1)
- [x] Build the package locally using the 2.8.0 app
  - [x] RPM
  - [x] DEB
- [ ] ~~Test the package (CentOS/Debian)~~
   - [ ] ~~Install~~
   - [ ] ~~Upgrade~~
   - [ ] ~~Remove~~
   - [ ] ~~Logs~~
   - [ ] ~~Alerts~~
   - [ ] ~~Versions references~~
- [x] Modify https://github.com/wazuh/wazuh-packages/blob/master/README.md matrix with the new version 
- [x] Test the package using the Test_stack pipeline

## Validation

- [ ] The package presents normal operation and without errors

---

### Working branch

- https://github.com/wazuh/wazuh-packages/tree/enhancement/2392-bump-wazuh-dashboard-2.8.0-4.6.0

**Comment by juliamagan** [comment]
- Blocked by https://github.com/wazuh/wazuh-packages/issues/2393:
  ```
  Updating files: 100% (9491/9491), done.
  + mkdir -p plugin_platform_source/plugins
  + mv plugin_platform_source /tmp/source
  + install_dependencies
  + cd /tmp/source
  + change_node_version '>=14.20.1' '<19'
  ++ node -v
  + installed_node_version=v10.24.1
  + node_version='>=14.20.1'
  + n '>=14.20.1'
  
    Error: invalid version '>=14.20.1'
```

**Comment by rauldpm** [comment]
## Update report

- Creating a Wazuh dashboard app in warehouse-test bucket with revision wp.2392

**Comment by rauldpm** [comment]
## Update report

- The Wazuh dashboard packages have been build using the Packages_builder pipeline, this issue changes to On hold until the issue https://github.com/wazuh/wazuh-jenkins/issues/5596 is completed and the Stack can be tested with the Test_stack pipeline

**Comment by rauldpm** [comment]
## Update report

- Building new packages for final testing

**Comment by rauldpm** [comment]
## Update report

- Changed ETA from 09/01/2023 to 09/05/2023 due to password change and template errors

**Comment by Deblintrake09** [comment]
## Update  report
- The documentation for the deprecation of IP usage in the configuration, states that FQDNs should be used instead. when trying to use a FQDN in the configuration the connection to the indexer was refused:

#### Configuration block
```
server.host: 0.0.0.0
opensearch.hosts: https://FQDN:9200/
server.port: 443
opensearch.ssl.verificationMode: certificate
# opensearch.username: kibanaserver
# opensearch.password: kibanaserver
opensearch.requestHeadersAllowlist: ["securitytenant","Authorization"]
opensearch_security.multitenancy.enabled: false
opensearch_security.readonly_mode.roles: ["kibana_read_only"]
server.ssl.enabled: true
server.ssl.key: "/etc/wazuh-dashboard/certs/wazuh-dashboard-key.pem"
server.ssl.certificate: "/etc/wazuh-dashboard/certs/wazuh-dashboard.pem"
opensearch.ssl.certificateAuthorities: ["/etc/wazuh-dashboard/certs/root-ca.pem"]
uiSettings.overrides.defaultRoute: /app/wazuh
opensearch_security.cookie.secure: true
```
#### Connection error found on restart
```
sep 04 22:34:15 ip-172-31-15-4.ec2.internal systemd[1]: Started wazuh-dashboard.
sep 04 22:34:15 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: v16.20.0
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","plugins-service"],"pid":11411,"message":"Plugin \"dataSourceManagement\" has been disabled since the following direct or transitive dependencies are missing or disabled: [dataSource]"}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","plugins-service"],"pid":11411,"message":"Plugin \"dataSource\" is disabled."}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","plugins-service"],"pid":11411,"message":"Plugin \"visTypeXy\" is disabled."}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","plugins-service"],"pid":11411,"message":"Plugin \"mlCommonsDashboards\" is disabled."}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","plugins-system"],"pid":11411,"message":"Setting up [44] plugins: [alertingDashboards,usageCollection,opensearchDashboardsUsageCollection,opensearchDashboardsLegacy,mapsLegacy,share,opensearchUiShared,legacyExport,embeddable,expressions,data,home,console,apmOss,management,indexPatternManagement,advancedSettings,savedObjects,reportsDashboards,indexManagementDashboards,dashboard,visualizations,visTypeVega,visTypeTimeline,visTypeTable,visTypeMarkdown,visBuilder,tileMap,regionMap,customImportMapDashboards,inputControlVis,ganttChartDashboards,visualize,notificationsDashboards,charts,visTypeVislib,visTypeTimeseries,visTypeTagcloud,visTypeMetric,discover,savedObjectsManagement,securityDashboards,wazuh,bfetch]"}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["info","savedobjects-service"],"pid":11411,"message":"Waiting until all OpenSearch nodes are compatible with OpenSearch Dashboards before starting saved objects migrations..."}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["error","opensearch","data"],"pid":11411,"message":"[ConnectionError]: connect ECONNREFUSED 172.31.15.4:9200"}
sep 04 22:34:18 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:18Z","tags":["error","savedobjects-service"],"pid":11411,"message":"Unable to retrieve version information from OpenSearch nodes."}
sep 04 22:34:21 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:21Z","tags":["error","opensearch","data"],"pid":11411,"message":"[ConnectionError]: connect ECONNREFUSED 172.31.15.4:9200"}
sep 04 22:34:23 ip-172-31-15-4.ec2.internal opensearch-dashboards[11411]: {"type":"log","@timestamp":"2023-09-04T22:34:23Z","tags":["error","opensearch","data"],"pid":11411,"message":"[ConnectionError]: connect ECONNREFUSED 172.31.15.4:9200"}
```

- While trying to apply the workaround to ignore deprecation warning, adding the `--no-warnings` option in the script `/usr/share/wazuh-dashboard/bin/opensearch-dashboards` used by the wazuh-dashboard service, the deprecation warning was still present.

```
sep 04 22:45:25 ip-172-31-15-4.ec2.internal systemd[1]: Started wazuh-dashboard.
sep 04 22:45:25 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: v16.20.0
sep 04 22:45:28 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:28Z","tags":["info","plugins-service"],"pid":11533,"message":"Plugin \"dataSourceManagement\" has been disabled since the following direct or transitive dependencies are missing or disabled: [dataSource]"}
sep 04 22:45:28 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:28Z","tags":["info","plugins-service"],"pid":11533,"message":"Plugin \"dataSource\" is disabled."}
sep 04 22:45:28 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:28Z","tags":["info","plugins-service"],"pid":11533,"message":"Plugin \"visTypeXy\" is disabled."}
sep 04 22:45:28 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:28Z","tags":["info","plugins-service"],"pid":11533,"message":"Plugin \"mlCommonsDashboards\" is disabled."}
sep 04 22:45:28 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:28Z","tags":["info","plugins-system"],"pid":11533,"message":"Setting up [44] plugins: [alertingDashboards,usageCollection,opensearchDashboardsUsageCollection,opensearchDashboardsLegacy,mapsLegacy,share,opensearchUiShared,legacyExport,embeddable,expressions,data,home,console,apmOss,management,indexPatternManagement,advancedSettings,savedObjects,reportsDashboards,indexManagementDashboards,dashboard,visualizations,visTypeVega,visTypeTimeline,visTypeTable,visTypeMarkdown,visBuilder,tileMap,regionMap,customImportMapDashboards,inputControlVis,ganttChartDashboards,visualize,notificationsDashboards,charts,visTypeVislib,visTypeTimeseries,visTypeTagcloud,visTypeMetric,discover,savedObjectsManagement,securityDashboards,wazuh,bfetch]"}
sep 04 22:45:29 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:29Z","tags":["info","savedobjects-service"],"pid":11533,"message":"Waiting until all OpenSearch nodes are compatible with OpenSearch Dashboards before starting saved objects migrations..."}
sep 04 22:45:29 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:29Z","tags":["info","savedobjects-service"],"pid":11533,"message":"Starting saved objects migrations"}
sep 04 22:45:29 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:29Z","tags":["info","plugins-system"],"pid":11533,"message":"Starting [44] plugins: [alertingDashboards,usageCollection,opensearchDashboardsUsageCollection,opensearchDashboardsLegacy,mapsLegacy,share,opensearchUiShared,legacyExport,embeddable,expressions,data,home,console,apmOss,management,indexPatternManagement,advancedSettings,savedObjects,reportsDashboards,indexManagementDashboards,dashboard,visualizations,visTypeVega,visTypeTimeline,visTypeTable,visTypeMarkdown,visBuilder,tileMap,regionMap,customImportMapDashboards,inputControlVis,ganttChartDashboards,visualize,notificationsDashboards,charts,visTypeVislib,visTypeTimeseries,visTypeTagcloud,visTypeMetric,discover,savedObjectsManagement,securityDashboards,wazuh,bfetch]"}
sep 04 22:45:29 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:29Z","tags":["listening","info"],"pid":11533,"message":"Server running at https://0.0.0.0:443"}
sep 04 22:45:29 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: {"type":"log","@timestamp":"2023-09-04T22:45:29Z","tags":["info","http","server","OpenSearchDashboards"],"pid":11533,"message":"http server running at https://0.0.0.0:443"}
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: (node:11533) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: (Use `node --trace-deprecation ...` to show where the warning was created)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: Node.js process-warning detected:
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at Object.connect (node:_tls_wrap:1678:15)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at HttpsAgent.createConnection (node:https:147:22)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at HttpsAgent.createSocket (/usr/share/wazuh-dashboard/node_modules/agentkeepalive/lib/_http_agent.js:265:26)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at HttpsAgent.createSocket (/usr/share/wazuh-dashboard/node_modules/agentkeepalive/lib/agent.js:77:11)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at HttpsAgent.addRequest (/usr/share/wazuh-dashboard/node_modules/agentkeepalive/lib/_http_agent.js:239:10)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at new ClientRequest (node:_http_client:335:16)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at Object.request (node:https:357:10)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at HttpConnector.request (/usr/share/wazuh-dashboard/node_modules/elasticsearch/src/lib/connectors/http.js:182:23)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at sendReqWithConnection (/usr/share/wazuh-dashboard/node_modules/elasticsearch/src/lib/transport.js:263:35)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at Object.utils.applyArgs (/usr/share/wazuh-dashboard/node_modules/elasticsearch/src/lib/utils.js:188:19)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at wrapper (/usr/share/wazuh-dashboard/node_modules/lodash/lodash.js:5255:19)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]:     at processTicksAndRejections (node:internal/process/task_queues:78:11)
sep 04 22:46:05 ip-172-31-15-4.ec2.internal opensearch-dashboards[11533]: Terminating process...
sep 04 22:46:05 ip-172-31-15-4.ec2.internal systemd[1]: wazuh-dashboard.service: Main process exited, code=exited, status=1/FAILURE
sep 04 22:46:05 ip-172-31-15-4.ec2.internal systemd[1]: wazuh-dashboard.service: Failed with result 'exit-code'.
```

**Comment by rauldpm** [comment]
## Update report

- The error related to not being able to use IP in the OpenSearch configuration will be investigated by @wazuh/frontend, in the meantime, names should be used

**Comment by uhlhosting** [comment]
Guys...

This is terrible!

Version 2.6.0 Its since February 2023.

Latest current release its on 2.11.0, and soon to be 2.12.0 .

Yet I see here you guys are stuck at version 2.8.0 . 

May I ask, what is going on? 

Have you had issues with https://github.com/opensearch-project/opensearch-build ? Address them there, do not chase rabbits here while it seems you go in circles. 

A bit more than shameful for a security stack, to be so much left behind in versioning with one of the most crucial part of the whole product. Beside OSSEC...
